stdlib/public/core/ObjectIdentifier.swift

```diff
@@ -14,6 +14,11 @@
 ///
 /// In Swift, only class instances and metatypes have unique identities. There
 /// is no notion of identity for structs, enums, functions, or tuples.
+///
+/// `ObjectIdentifier` is only guaranteed to remain unique for the
+/// lifetime of an object. When the instance gets deallocated, its object 
+/// identifier may be reused for a different object. (Internally, objects are
+/// identified by their memory location.)
```

Comment:
What does this parenthetical mean, and why is it important for users to know this?

Reply:
As I mentioned on the https://bugs.swift.org/browse/SR-13564, the result of this example is not as expected.

```
   func makeIdentifier(input: Int) -> ObjectIdentifier {
        ObjectIdentifier(NSNumber(value:input));
    }
    
    let a = makeIdentifier(input: Int.min);
    let b = makeIdentifier(input: Int.max);
    print(a == b) // print: true

```

Reply:
You're right, that result _would_ be unexpected. However, I just ran your example, and on my machine it prints `false`, which is what I would expect.

Under what conditions are you getting `true`? I'm not sure I would understand how that comes about, and the parenthetical doesn't clarify for me.

Reply:
That example could give different results, depending on whether:

* tagged pointers are supported (Apple platforms only), or
* the allocator decides to reuse the same memory location.

(Although tagged pointers might not be used for `Int.min` and `Int.max`).

Reply:
> You're right, that result _would_ be unexpected. However, I just ran your example, and on my machine it prints `false`, which is what I would expect.
> 
> Under what conditions are you getting `true`? I'm not sure I would understand how that comes about, and the parenthetical doesn't clarify for me.

Mac x86_64 & Xcode